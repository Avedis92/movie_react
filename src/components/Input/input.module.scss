@import "../sasscommonmixxins/flex.scss";
@import "../sasscommonmixxins/mediaqueries.scss";

.searchHeader {
  @include flexElements(flex, row, center, space-between);
  background-color: rgba(0, 0, 0, 0.486);
  width: 100%;
  height: 50px;
  position: fixed;
  z-index: 1;
  color: White;
  h1 {
    margin-left: 5%;
    cursor: pointer;
    transition: 500ms;
    &:hover {
      transform: scale(105%);
      color: rgb(0, 195, 255);
    }
    @include mobile {
      margin-left: 0;
      font-size: 26px;
    }
  }
  input {
    margin-right: 5%;
    width: 20%;
    border-color: white;
    border-radius: 10px;
    height: 70%;
    background-color: transparent;
    color: white;
    font-style: italic;
    font-size: 18px;
    @include mobile {
      width: 50vw;
      font-size: 16px;
    }
  }
  ::placeholder {
    color: white;
    opacity: 1;
    font-style: italic;
    font-size: 18px;
    @include mobile {
      font-size: 16px;
    }
  }
}
