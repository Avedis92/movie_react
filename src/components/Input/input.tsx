import { useRef } from 'react';
import { useHistory } from 'react-router-dom';
import styles from './input.module.scss';

function Input({ setupMoviePage }:{setupMoviePage?:()=>void}) {
  const history = useHistory();
  const inputElem = useRef<HTMLInputElement>(null);
  const { searchHeader } = styles;
  const PressEnter = (e:React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter' && inputElem.current !== null) {
      if (setupMoviePage) {
        setupMoviePage();
      }
      history.push(`/search?searchedMovie=${inputElem.current.value}`);
    }
  };
  return (
    <div className={searchHeader}>
      <h1 onClick={() => history.push('/')}>Movies</h1>
      <input
        list="movieTitle"
        type="text"
        name="movieSearch"
        onKeyPress={PressEnter}
        placeholder="Search"
        ref={inputElem}
      />
    </div>
  );
}

export default Input;
