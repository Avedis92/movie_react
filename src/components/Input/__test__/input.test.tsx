import { render } from '@testing-library/react';
import Input from '../input';

test('Create a snapshot test', () => {
  const {asFragment} = render(<Input setupMoviePage={() => null} />);
  expect(asFragment()).toMatchSnapshot();
});
