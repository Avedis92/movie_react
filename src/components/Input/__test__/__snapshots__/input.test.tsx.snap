// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="searchHeader"
  >
    <h1>
      Movies
    </h1>
    <input
      list="movieTitle"
      name="movieSearch"
      placeholder="Search"
      type="text"
    />
  </div>
</DocumentFragment>
`;
